src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -873,3 +1049,93 @@ _ostree_repo_static_delta_dump (OstreeRepo                    *self,
 
   return TRUE;
 }
+
+/**
+ * ostree_repo_static_delta_is_signed:
+ * @self: Repo
+ * @delta_id: delta path
+ * @out_value: (out) (element-type utf8) (transfer container): location to store
+ *            the list of signature type strings.
+ * @error: Error
+ *
+ * Verify if the static-delta has signature(s).
+ * If the option is not set, or if an error is returned, @out_value will be set
+ * to %NULL.
+ *
+ * Returns: TRUE if static data has signatures, FALSE otherwise.
+ *
+ * Since: 2020.1
+ */
+gboolean
+ostree_repo_static_delta_is_signed (OstreeRepo       *self,
```

Comment:
Looks like we don't need to expose this function anymore -- function `ostree_repo_static_delta_execute_offline_with_signature()` should to return correct error message in case if no signature found.

Reply:
I sent a new version with following changes:
- lib/deltas: Add signature check API for static-delta superblock
  remove public API `ostree_repo_static_delta_is_signed`
  moving `ostree_repo_static_delta_is_signed` code to static function and call it from `ostree_repo_static_delta_verify_signature` before trying to verify signature
- bin/static-delta: Add command to verify delta signature
  remove call to `ostree_repo_static_delta_is_signed` as it is done in `ostree_repo_static_delta_verify_signature`
- Update tests as `static-delta verify` call with unsigned delta now return failure

@d4s if you think it's ok I will sent this upstream